pkg/skaffold/kubernetes/watcher.go

```diff
@@ -114,6 +129,24 @@ func (w *podWatcher) Start(kubeContext string, namespaces []string) (func(), err
 					continue
 				}
 
+				if l.Logger.IsLevelEnabled(logrus.TraceLevel) {
```

Comment:
is this check `l.Logger.IsLevelEnabled(logrus.TraceLevel)` required? Doesn't `l.Trace` already handle it? Is it to save the cost of constructing the string?

Reply:
Yeah, probably premature optimization: I thought the podWatcher fired more events than it actually does.  This dereferencing seemed a bit distasteful but our output logger just returns a logrus object anyways.

Reply:
`pkg/skaffold/kubernetes/watcher.go:25:2: Do not use github.com/sirupsen/logrus package, use output.log.Entry instead` 😭